_posts/2019-03-21-timezone-in-rails-and-javascript.md

```diff
@@ -0,0 +1,96 @@
+  - database
+  - server
+  - dev machine
+  - system configured
+  - user specific configured
+  - the browser
+
+So what tools do we have at our disposal as Rails developers?
+
+```ruby
+config.time_zone = 'Eastern Time (US & Canada)' # rails application timezone
+config.active_record.default_timezone = :local  # controls the timezone datetime is converted to prior saving to DB
+```
+The most important one is the `config.time_zone` configuration in your __*config/application.rb*__ file. This sets the application timezone. ActiveRecord will help you convert from and to UTC and the timezone of your choice.
+
+Another one is `config.active_record.default_timezone` configuration in your __*config/application.rb*__ file, This controls in which timezone DB entries are treated at. It can be either UTC or Local. And UTC is default. If you set it to local, ActiveRecord will help you convert from and to system local and the timezone of your choice when setting and retrieving from the database.
+
+<div class="information information-warning">
+  In almost all scenarios, we don't want to change this value, and we want to keep UTC as the default timezone for Database.
+</div>
+
+If all you're doing is receiving user data through a form and use Active Record to persist it you're good to go. But what about actually doing something with the time information before persisting it? That's when it becomes tricky.
+
+When parsing time information it's important to never do it without specifying the timezone. The best way to do this is to use the methods that use the timezone specified in our rails application config. Let's look at a couple of methods that parses date and time:
+
+**Methods**                           | **Description**
+===================================== | ===============
+`Time.current`                        | uses timezone from rails application config
+`Time.now`                            | uses timezone from system locale
+`Time.zone.xxx`                       | uses timezone from rails application config
+`Time.strptime(xx, xx).in_time_zone`  | uses timezone from rails application config
+`Time.strptime(xx, xx)`               | uses timezone from system locale
+`Date.current`                        | uses timezone from rails application config
+`Date.today`                          | uses timezone from system locale
+`String.to_time`                      | uses timezone from system locale
+`String.in_time_zone`                 | uses timezone from rails application config
+
+Looking carefully, you'll discover that the naming of the methods that adopt rails application configured timezone has a set of common patterns:
+1. we should use the `.current` method whenever available
+2. we should use the `zone` object as intermediate object whenever available
+3. we could use `.in_time_zone` on any value to reset its timezone to rails application config
```

Comment:
These are a great set of rules! I might be tempted to put these in a section or somehow make this stand out a bit more.

Reply:
Great! I make it pop with some styling, do you like it: 
![image](https://user-images.githubusercontent.com/871037/55356468-c8ac2d80-5498-11e9-817e-2a9332474217.png)

Reply:
@JJPro nice! Just need to wrap those backticks in `<code>` tags manually now it seems.

Reply:
good catch 🎉